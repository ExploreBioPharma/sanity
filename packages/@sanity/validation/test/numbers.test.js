const {Rule} = require('../src')

describe('number', () => {
  test('required constraint', () => {
    const rule = Rule.number().required()
    expect(rule.validate(undefined)).toMatchSnapshot('required: undefined')
    expect(rule.validate(13)).toMatchSnapshot('required: valid')
  })

  test('min constraint', () => {
    const rule = Rule.number().min(10)
    expect(rule.validate(3)).toMatchSnapshot('min: too low')
    expect(rule.validate(10)).toMatchSnapshot('min: valid, at limit')
    expect(rule.validate(20)).toMatchSnapshot('min: valid')
  })

  test('greater than constraint', () => {
    const rule = Rule.number().greaterThan(10)
    expect(rule.validate(3)).toMatchSnapshot('gt: too low')
    expect(rule.validate(10)).toMatchSnapshot('gt: too low, at limit')
    expect(rule.validate(20)).toMatchSnapshot('gt: valid')
  })

  test('max constraint', () => {
    const rule = Rule.number().max(10)
    expect(rule.validate(20)).toMatchSnapshot('max: too large')
    expect(rule.validate(10)).toMatchSnapshot('max: valid, at limit')
    expect(rule.validate(5)).toMatchSnapshot('max: valid')
  })

  test('less than constraint', () => {
    const rule = Rule.number().lessThan(10)
    expect(rule.validate(20)).toMatchSnapshot('lt: too high')
    expect(rule.validate(10)).toMatchSnapshot('lt: too high, at limit')
    expect(rule.validate(3)).toMatchSnapshot('lt: valid')
  })

  test('integer constraint', () => {
    const rule = Rule.number().integer()
    expect(rule.validate(31.14)).toMatchSnapshot('integer: invalid (float)')
    expect(rule.validate(31)).toMatchSnapshot('integer: valid')
  })

  test('precision constraint', () => {
    const rule = Rule.number().precision(3)
    expect(rule.validate(Math.PI)).toMatchSnapshot('precision: invalid (pi)')
    expect(rule.validate(31.133)).toMatchSnapshot('precision: valid (at limit)')
    expect(rule.validate(31.3)).toMatchSnapshot('precision: valid (below limit)')
  })

  test('positive constraint', () => {
    const rule = Rule.number().positive()
    expect(rule.validate(-31.14)).toMatchSnapshot('positive: invalid')
    expect(rule.validate(0)).toMatchSnapshot('positive: valid (zero)')
    expect(rule.validate(13)).toMatchSnapshot('positive: valid')
  })

  test('negative constraint', () => {
    const rule = Rule.number().negative()
    expect(rule.validate(31.14)).toMatchSnapshot('negative: invalid')
    expect(rule.validate(0)).toMatchSnapshot('negative: invalid (zero)')
    expect(rule.validate(-13)).toMatchSnapshot('negative: valid')
  })
})
