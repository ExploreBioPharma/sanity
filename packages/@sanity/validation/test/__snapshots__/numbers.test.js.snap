// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gt: too low 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be greater than 10],
    "level": "error",
  },
]
`;

exports[`gt: too low, at limit 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be greater than 10],
    "level": "error",
  },
]
`;

exports[`gt: valid 1`] = `Array []`;

exports[`integer: invalid (float) 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be an integer],
    "level": "error",
  },
]
`;

exports[`integer: valid 1`] = `Array []`;

exports[`lt: too high 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be less than 10],
    "level": "error",
  },
]
`;

exports[`lt: too high, at limit 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be less than 10],
    "level": "error",
  },
]
`;

exports[`lt: valid 1`] = `Array []`;

exports[`max: too large 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be less than or equal 10],
    "level": "error",
  },
]
`;

exports[`max: valid 1`] = `Array []`;

exports[`max: valid, at limit 1`] = `Array []`;

exports[`min: too low 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be greater than or equal 10],
    "level": "error",
  },
]
`;

exports[`min: valid 1`] = `Array []`;

exports[`min: valid, at limit 1`] = `Array []`;

exports[`negative: invalid (zero) 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be less than 0],
    "level": "error",
  },
]
`;

exports[`negative: invalid 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be less than 0],
    "level": "error",
  },
]
`;

exports[`negative: valid 1`] = `Array []`;

exports[`positive: invalid 1`] = `
Array [
  Object {
    "item": [ValidationError: Number must be greater than or equal 0],
    "level": "error",
  },
]
`;

exports[`positive: valid (zero) 1`] = `Array []`;

exports[`positive: valid 1`] = `Array []`;

exports[`precision: invalid (pi) 1`] = `
Array [
  Object {
    "item": [ValidationError: Max precision is 3],
    "level": "error",
  },
]
`;

exports[`precision: valid (at limit) 1`] = `Array []`;

exports[`precision: valid (below limit) 1`] = `Array []`;

exports[`required: undefined 1`] = `
Array [
  Object {
    "item": [ValidationError: Value is required],
    "level": "error",
  },
]
`;

exports[`required: valid 1`] = `Array []`;
